doxygen/contributor_help_pages/common_pitfalls.md

```diff
@@ -191,43 +188,42 @@ class my_big_type {
 
   explicit my_big_type(size_t N) m_data(static_cast<double*>(malloc(N))) {}
   // Standard copy constructor
-  my_big_type(const my_big_type& other) m_data(static_cast<double*>(malloc(N))) {
+  my_big_type(const my_big_type& other) : m_data(static_cast<double*>(malloc(N))) {
     std::copy(other.m_data, this->m_data);
   }
   // Standard move constructor
-  my_big_type(my_big_type&& other) m_data(other.m_data) {
+  my_big_type(my_big_type&& other) : m_data(other.m_data) {
     other.m_data = nullptr;
   }
 };
 ```
 
 We can see in the above that the standard style of a move (the constructor taking an rvalue reference) is to copy the pointer and then null out the original pointer. But in Stan, particularly for reverse mode, we need to keep memory around even if  it's only a temporary for when we call the gradient calculations in the reverse pass. And since memory for reverse mode is stored in our arena allocator no copying happens in the first place.
 
-When working with arithmetic types, keep in mind that moving Scalars is often less optimal than simply taking their copy. For instance, Stan's `var` type is a PIMPL implementation, so it simply holds a pointer of size 8 bytes. A `double` is also 8 bytes which just so happens to fit exactly in a [word](https://en.wikipedia.org/wiki/Word_(computer_architecture)) of most modern CPUs. While a reference to a double is also 8 bytes, unless the function is inlined by the compiler, the computer will have to place the reference into cache, then go fetch the value that is being referenced which now takes up two words instead of one!
+When working with arithmetic types, keep in mind that moving Scalars is often less optimal than simply taking their copy. For instance, Stan's `var` type is a PIMPL implementation, so it simply holds a pointer of size 8 bytes. A `double` is also 8 bytes which just so happens to fit exactly in a [word](https://en.wikipedia.org/wiki/Word_(computer_architecture)) of most modern CPUs with at least 64 byte cache lines. While a reference to a double is also 8 bytes, unless the function is inlined by the compiler, the computer will have to place the reference into cache, then go fetch the value that is being referenced which now takes up two words instead of one!
```

Comment:
> I think we should avoid these super long lines.

Yeah I'll add some returns here to make the markdown easier to read.

> But the reason I opened this box was to say that "PIMPL impementation" is redundant because the IMPL stands for "implementation. So I'd say "uses the pointer to implementation (PIMPL) pattern" to avoid redundancy.

Yes thats a much better way to word it, fixed!

> Plus, thanks for explaining the "why" here---I hadn't realized references would tickle two words instead of one. I would've expected that with a pointer, but I would've thought references would be smarter given that they're immutable.

Yes, though note that this is one of those weird internal to compiler things that tbh will rarely come up, but with large programs you can have instances where this sort of thing happens.